### PR TITLE
VDS-252 Split product page to separate snippets

### PR DESCRIPTION
**Pull request by trueboroda**
Physical with variations

Physical without variations

Configured product

DoD:

* Each snippet handle only self product type
* If something could be reused, it should be reused
* Separate angular controller
* Product card (controller only)

### CODE REVIEW

File: assets/js/controllers/product-card/common-product-card.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            event.preventDefault();
+            catalogService.getProduct([productId]).then(function (response) {
+                if (response.data && response.data.length) {
+                    var product = response.data[0];
+                    $scope.addProductToCart(product, quantity);
+                }
+            });
+        }
+
+        $scope.sendToEmail = function (storeId, productId, productUrl, language) {
```
Comment (trueboroda) [inline assets/js/controllers/product-card/common-product-card.js:28]: rename To something like 'showAddedDialog'

File: snippets/product/type/without-variations/product.liquid [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            </li>
+            <li>
+                <span class="text-muted">{{ 'products.product.mmn' | t }}</span>&nbsp;<strong itemprop="model">{{ product.properties['MMN'].value }}</strong>
+            </li>
+            <li>
+                <span class="text-muted">UNSPSC</span>&nbsp;<strong>#41112701</strong>
+            </li>
+        </ul>
+    </div>
+</div>
```
Comment (asvishnyakov) [inline snippets/product/type/without-variations/product.liquid:17]: This part looks exactly the same as for other types. May we reuse the code?

File: snippets/product/type/without-variations/product.liquid [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                        <p ng-show="shipmentType == 'shipping' && shipmentAddress">Expected to arrive: <strong ng-bind="availability['{{ product.id }}'].expectedArrival | date: 'fullDate'"></strong></p>
+                        <p ng-show="shipmentType == 'pickup' && shipmentFulfillmentCenter">Available for pickup: <strong ng-bind="availability['{{ product.id }}'].availableSince | date: 'fullDate'"></strong></p>
+                        <vc-payment-plan available-payment-plans="::availablePaymentPlans"></vc-payment-plan>
+                        <div class="form-group">
+                            <ng-controller ng-controller="commonProductCardController">
+                                {% include 'product/quantity' %}
+                            </ng-controller>
+                        </div>
+                    </div>
+                </div>
```
Comment (asvishnyakov) [inline snippets/product/type/without-variations/product.liquid:59]: This part looks exactly the same as for other types. May we reuse the code?

File: snippets/product/type/with-variations/product.liquid [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                        <p ng-show="shipmentType == 'shipping' && shipmentAddress">Expected to arrive: <strong ng-bind="availability['{{ product.id }}'].expectedArrival | date: 'fullDate'"></strong></p>
+                        <p ng-show="shipmentType == 'pickup' && shipmentFulfillmentCenter">Available for pickup: <strong ng-bind="availability['{{ product.id }}'].availableSince | date: 'fullDate'"></strong></p>
+                        <vc-payment-plan available-payment-plans="::availablePaymentPlans"></vc-payment-plan>
+                        <div class="form-group">
+                            <ng-controller ng-controller="commonProductCardController">
+                                {% include 'product/quantity' %}
+                            </ng-controller>
+                        </div>
+                    </div>
+                </div>
```
Comment (asvishnyakov) [inline snippets/product/type/with-variations/product.liquid:59]: This part looks exactly the same as for other types. May we reuse the code?

File: snippets/product/type/with-variations/product.liquid [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            </li>
+            <li>
+                <span class="text-muted">{{ 'products.product.mmn' | t }}</span>&nbsp;<strong itemprop="model">{{ product.properties['MMN'].value }}</strong>
+            </li>
+            <li>
+                <span class="text-muted">UNSPSC</span>&nbsp;<strong>#41112701</strong>
+            </li>
+        </ul>
+    </div>
+</div>
```
Comment (asvishnyakov) [inline snippets/product/type/with-variations/product.liquid:17]: This part looks exactly the same as for other types. May we reuse the code?

File: snippets/product/type/configured/product.liquid [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+                <div class="delivery-cart">
+                    <div ng-controller="purchaseController" ng-init='loadPaymentPlan({{ settings.available_paymentPlans | json }}, "product", "{{ product.id }}")'>
+                        <p ng-show="shipmentType == 'shipping' && shipmentAddress">Expected to arrive: <strong ng-bind="availability['{{ product.id }}'].expectedArrival | date: 'fullDate'"></strong></p>
+                        <p ng-show="shipmentType == 'pickup' && shipmentFulfillmentCenter">Available for pickup: <strong ng-bind="availability['{{ product.id }}'].availableSince | date: 'fullDate'"></strong></p>
+                        <vc-payment-plan available-payment-plans="::availablePaymentPlans"></vc-payment-plan>
+                        <div class="form-group">
+                            {% include 'product/type/configured/quantity' %}
+                        </div>
+                    </div>
+                </div>
```
Comment (asvishnyakov) [inline snippets/product/type/configured/product.liquid:96]: This part looks exactly the same as for other types. May we reuse the code?

File: snippets/product/type/configured/product.liquid [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            </li>
+            <li>
+                <span class="text-muted">{{ 'products.product.mmn' | t }}</span>&nbsp;<strong itemprop="model">{{ product.properties['MMN'].value }}</strong>
+            </li>
+            <li>
+                <span class="text-muted">UNSPSC</span>&nbsp;<strong>#41112701</strong>
+            </li>
+        </ul>
+    </div>
+</div>
```
Comment (asvishnyakov) [inline snippets/product/type/configured/product.liquid:16]: This part looks similar to other types. May we reuse the code?

File: snippets/product/type/configured/item-configure-button.liquid [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    {% if unavailable %}
+    {{ 'products.product.unavailable' | t }}
+    {% elsif sold_out %}
+    {{ 'products.product.sold_out' | t }}
+    {% else %}
+    <img class="img-responsive configure-icon" src="{{ 'customize.svg' | asset_url }}" />
+    <span>Configure</span>
+    {% endif %}
+    </button>
+</a>
```
Comment (asvishnyakov) [inline snippets/product/type/configured/item-configure-button.liquid:14]: May we use any formatting tool (for example, extension for VS Code) to format Liquid and indent `if-else` and similar blocks?

File: snippets/product/quantity.liquid [hunk 9]
```diff
@@ -4,30 +4,22 @@
...
-                    {{ 'products.general.add_to_cart' | t }}
-                    {% endif %}
+                    >
+                        {% if unavailable %}
+                        {{ 'products.product.unavailable' | t }}
+                        {% elsif sold_out %}
+                        {{ 'products.product.sold_out' | t }}
+                        {% else %}
+                        {{ 'products.general.add_to_cart' | t }}
+                        {% endif %}
```
Comment (asvishnyakov) [inline snippets/product/quantity.liquid:22]: May we use any formatting tool (for example, extension for VS Code) to format Liquid and indent if-else and similar blocks?

**Review by asvishnyakov — CHANGES_REQUESTED** [review]
I reviewed and wrote all comments I wanted. As we discussed at one of daily meetings, we probably may want to reuse a part of controllers code and I will refactor it to show what I mean. Please let me do it a little bit later

File: snippets/product/type/without-variations/product.liquid (hunk 2 above)
Comment (ilyawzrd) [reply snippets/product/type/without-variations/product.liquid]: done

File: snippets/product/type/with-variations/product.liquid (hunk 5 above)
Comment (ilyawzrd) [reply snippets/product/type/with-variations/product.liquid]: done

File: snippets/product/type/configured/product.liquid (hunk 7 above)
Comment (ilyawzrd) [reply snippets/product/type/configured/product.liquid]: done

File: snippets/product/type/configured/product.liquid (hunk 6 above)
Comment (ilyawzrd) [reply snippets/product/type/configured/product.liquid]: done

File: snippets/product/type/with-variations/product.liquid (hunk 4 above)
Comment (ilyawzrd) [reply snippets/product/type/with-variations/product.liquid]: done

File: snippets/product/type/without-variations/product.liquid (hunk 3 above)
Comment (ilyawzrd) [reply snippets/product/type/without-variations/product.liquid]: done

File: snippets/product/type/configured/item-configure-button.liquid (hunk 8 above)
Comment (ilyawzrd) [reply snippets/product/type/configured/item-configure-button.liquid]: done

File: snippets/product/quantity.liquid (hunk 9 above)
Comment (ilyawzrd) [reply snippets/product/quantity.liquid]: done

**Review by asvishnyakov — APPROVED** [review]

**Review by pushnitsa — APPROVED** [review]

**Pull request merged**